### PR TITLE
Update fu-rts54hub-device.c for gen1 gen2 hub status response length

### PR DESCRIPTION
**Pull request by AnyProblem**
gen2 hub get status max length is 25
gen1 hub get status max length is 24 

In order to cover gen1 and gen2 hub FW update so changed 25 to 24. Length 24 is enough to get information from gen1 and gen2 hub

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ *] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: plugins/rts54hub/fu-rts54hub-device.c [hunk 1]
```diff
@@ -25,7 +25,7 @@ G_DEFINE_TYPE(FuRts54HubDevice, fu_rts54hub_device, FU_TYPE_USB_DEVICE)
 #define FU_RTS54HUB_DEVICE_TIMEOUT_ERASE 5000  /* ms */
 #define FU_RTS54HUB_DEVICE_TIMEOUT_AUTH	 10000 /* ms */
 #define FU_RTS54HUB_DEVICE_BLOCK_SIZE	 4096
-#define FU_RTS54HUB_DEVICE_STATUS_LEN	 25
+#define FU_RTS54HUB_DEVICE_STATUS_LEN	 24
```
Comment (hughsie) [inline plugins/rts54hub/fu-rts54hub-device.c:28]: so this changes the size of the buffer -- this won't regress on older hardware or firmware versions?

Comment (AnyProblem) [reply plugins/rts54hub/fu-rts54hub-device.c:28]: The status of Gen2 hub, index [24] this BYTE we don't used on the update flow.
In order to this update flow on FWUPD can use on Gen1 and Gen2 hub of realtek  so changed STATUS-LEN to 24.
This change not affect older hardware or firmware versions.

**Review by hughsie — APPROVED** [review]

**Comment by hughsie** [comment]
Ohh the device-tests/realtek-rts5423.json emulated tests are now failing. We need to fix that.

**Comment by AnyProblem** [comment]
where is the device-tests/realtek-rts5423.json? what test on this json?

**Review by hughsie — CHANGES_REQUESTED** [review]
Out of interest; how can we tell if a hub is gen1 or gen2?

**Comment by AnyProblem** [comment]
We can't know the hub is whether gen1 or gen2 before get STATUS, this information hide in STATUS return,
if we want to know this before send get_status(), maybe need add gen1 or gen2 tag in quirk file by id to know

**Comment by hughsie** [comment]
> device-tests/realtek-rts5423.json

It's in fwupd -- and it points to a file on the LVFS. I've downloaded both emulation files, exploded them, manually truncated the data (e.g. `echo "C9oLAFsAgoD/AQFXBwNAAwAAAAABAAFXBg==" | base64 -d | head -c 24 | base64`) then re-zipped them, and uploaded them -- then modified the device test. It seems to work locally, lets see what CI says: https://github.com/fwupd/fwupd/pull/6988

**Pull request closed without merging**